### PR TITLE
Replace 'rm -rf' with 'rimraf' for cross-platform compatibility in package.json files

### PR DESCRIPTION
**Pull request by jarick**
### Summary

Replaces shell-specific `rm -rf` usage with `rimraf` to improve cross-platform compatibility, especially on Windows environments where Unix shell commands may not be available.

### Type of Change

* [x] 🐛 Bug fix (non-breaking change that fixes an issue)
* [ ] ✨ New feature (non-breaking change that adds functionality)
* [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] 📚 Documentation update
* [ ] 🔧 Code refactoring (no functional changes)
* [ ] ⚡ Performance improvement
* [ ] 🧪 Test improvements
* [x] 🏗️ Build system changes
* [ ] 🔒 Security fix
* [ ] 🎨 Style/formatting changes
* [x] 🔄 Dependency updates

### Related Issues

N/A

### Motivation and Context

The previous implementation used `rm -rf`, which depends on Unix-like shell utilities and can fail on Windows or other environments where this command is unavailable. Replacing it with `rimraf` provides a Node.js-based, cross-platform alternative for removing files and directories, making the build and cleanup scripts more reliable across different operating systems.

### Code Changes

* Replaced direct `rm -rf` command usage with `rimraf`.
* Updated cleanup/build scripts to use a cross-platform removal command.
* Ensured existing script behavior remains unchanged while improving OS compatibility.

### API Changes

N/A — no public API changes.

### Configuration Changes

* Added or updated the `rimraf` dependency where required.
* Updated package scripts/configuration to call `rimraf` instead of `rm -rf`.

### Test Coverage

* [ ] Unit tests added/updated
* [ ] Integration tests added/updated
* [ ] E2E tests added/updated
* [x] Manual testing performed
* [ ] Performance benchmarks run
* [x] Cross-platform testing completed

### Test Results

```bash
pnpm install
pnpm build
pnpm test
```

Build and test commands completed successfully locally.

### Manual Testing Steps

1. Installed dependencies to ensure `rimraf` is available.
2. Ran the affected cleanup/build script.
3. Verified that generated output/cache directories are removed correctly without relying on `rm -rf`.
4. Ran the project build/test command to confirm there are no regressions.

### Performance Considerations

* [x] No performance impact expected
* [ ] Performance improvement (provide metrics)
* [ ] Potential performance impact (explain and justify)
* [ ] Performance regression (explain mitigation)

### Benchmarks

N/A

### Breaking Changes Details

No breaking changes.

### Migration Guide

No migration required.

### Documentation Updates

* [ ] README updated
* [ ] API documentation updated
* [ ] Code comments added/updated
* [ ] Contributing guide updated
* [ ] Examples updated/added
* [ ] Changelog entry added
* [x] No documentation changes needed

### Documentation Links

N/A

### General

* [x] My code follows the project's style guidelines
* [x] I have performed a self-review of my code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [x] My changes generate no new warnings
* [ ] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes

### Rust-specific (if applicable)

* [ ] Code compiles without warnings (`cargo build`)
* [ ] All tests pass (`cargo test`)
* [ ] Code is properly formatted (`cargo fmt`)
* [ ] Clippy checks pass (`cargo clippy`)
* [ ] Public APIs are documented with doc comments
* [ ] Unsafe code is properly justified and documented

### TypeScript-specific (if applicable)

* [x] TypeScript compilation succeeds
* [x] ESLint checks pass
* [ ] Type definitions are accurate and complete
* [ ] JSDoc comments added for public APIs

### Build & Release

* [x] Build process works correctly
* [x] All platforms build successfully (if applicable)
* [ ] Version numbers updated (if applicable)
* [ ] Changelog updated (if applicable)

### Security

* [x] No sensitive information exposed in code or commits
* [x] Security implications have been considered
* [x] Dependencies are up to date and secure
* [ ] Input validation implemented where needed

### Screenshots

N/A

### Related Work

N/A

### Future Work

N/A

### Notes for Reviewers

Please focus on the updated package scripts and dependency changes to confirm that the cleanup behavior remains the same while working correctly across platforms.

## Summary by CodeRabbit

* **Chores**
  * Replaced platform-specific cleanup commands with a cross-platform cleaner across project scripts.
  * Added the cross-platform cleanup utility to devDependencies and registered it in the workspace catalog for templates and packages.

* **Tests**
  * Expanded unit tests to add coverage for additional export syntaxes and edge cases, improving build/test reliability.

### CODE REVIEW

File: packages/create-rari-app/package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
   "scripts": {
     "build": "pnpm clean && pnpm typecheck && vp pack",
     "dev": "vp pack --watch",
-    "clean": "rm -rf dist",
+    "clean": "rimraf dist",
```
[the remark this answers is not in the document]
Comment (jarick) [reply packages/create-rari-app/package.json:45]: fixed

**Comment by skiniks** [comment]
Thanks for this! `rimraf` isn't needed since we require Node.js >= 22.12.0, which has native support for recursive directory removal via `fs.rmSync()`.

Closing in favor of #203.
